src/Core/efcpt.8/HostedServices/ScaffoldHostedService.cs

```diff
@@ -84,6 +84,18 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
 #pragma warning restore S2583 // Conditionally executed code should be reachable
 #pragma warning restore S2589 // Boolean expressions should not be gratuitous
 
+        if (commandOptions.UseT4 && commandOptions.UseT4Split)
+        {
+            DisplayService.MarkupLine("Both UseT4 and UseT4Split are set to true.  Only one of thse should be used, UseT4Split will be ignored.", Color.Yellow);
+            commandOptions.UseT4Split = false;
+        }
+
+        if (commandOptions.UseT4Split && commandOptions.UseDbContextSplitting)
+        {
+            DisplayService.MarkupLine("Both UseDbContextSplitting and UseT4Split are set to true.  Only one of thse should be used, UseT4Split will be ignored.", Color.Yellow);
+            commandOptions.UseT4Split = false;
+        }
+
```

Comment:
is this where you'd want this or does it go somewhere else?

Reply:
Yes, this codepath is hit by both VS and the CLI, so a good place

Reply:
Sorry, missed the location while reviewing on my mobile, these checks should live in ReverseEngineerRunner (there are some checks there already) line 191